docs/migrations/migrations_guide.md

```diff
@@ -0,0 +1,192 @@
+# 📚 Database Migrations Guide – Nixopus
+
+This guide explains how database migrations work in Nixopus, how to create and apply them, and best practices for using them effectively.
+
+---
+
+## 🧰 Tool We Use: `golang-migrate`
+
+Nixopus uses [`golang-migrate`](https://github.com/golang-migrate/migrate) to manage SQL-based database migrations in Go. This tool supports programmatic and CLI-based migration execution, and we’ve embedded it into our application lifecycle so that migrations run automatically on startup.
+
+---
+
+## 📂 Where Migrations Live
+
+All migrations are stored in the `migrations/` directory (relative to the module where migration logic exists). Each migration consists of two SQL files:
```

Comment:
I think we should also add directory structure within directory folder.
Like if you see this `/migrations` folder:

- `auth/` - user auth, sesison & tokens
- `users/` - for user profile and settings
- `applications/` - app deployments, logs & status
- `organizations/` - org management
- `rbac/` - roles mgmt
- `domains/` - domain mgmt
- `notifications/` - email, webhooks etc
- `containers/` - container mgmt
- `integrations/` - 3rd service integrations 
- `audit/` - audit logs
- `feature-flags/` - feature flag management

If you see this structure, this is purely based on domain specific modular structuring. Let us add this information as well.